static/app/components/smartSearchBar/utils.tsx

```diff
@@ -160,3 +173,69 @@ export function filterSearchGroupsByIndex(items: SearchGroup[], index: number) {
 
   return foundSearchItem;
 }
+
+export function generateOperatorEntryMap(tag: string) {
+  return {
+    [TermOperator.Default]: {
+      type: 'tag-operator' as ItemType,
+      value: ':',
+      desc: `${tag}:${t('[value] is equal to')}`,
+    },
+    [TermOperator.GreaterThanEqual]: {
+      type: 'tag-operator' as ItemType,
+      value: ':>=',
+      desc: `${tag}:${t('>=[value] is greater than or equal to')}`,
+    },
+    [TermOperator.LessThanEqual]: {
+      type: 'tag-operator' as ItemType,
+      value: ':<=',
+      desc: `${tag}:${t('<=[value] is less than or equal to')}`,
+    },
+    [TermOperator.GreaterThan]: {
+      type: 'tag-operator' as ItemType,
+      value: ':>',
+      desc: `${tag}:${t('>[value] is greater than')}`,
+    },
+    [TermOperator.LessThan]: {
+      type: 'tag-operator' as ItemType,
+      value: ':<',
+      desc: `${tag}:${t('<[value] is less than')}`,
+    },
+    [TermOperator.Equal]: {
+      type: 'tag-operator' as ItemType,
+      value: ':=',
+      desc: `${tag}:${t('=[value] is equal to')}`,
+    },
+    [TermOperator.NotEqual]: {
+      type: 'tag-operator' as ItemType,
+      value: '!:',
+      desc: `!${tag}:${t('[value] is not equal to')}`,
+    },
+  };
+}
+
+export function getValidOps(
+  filterToken: TokenResult<Token.Filter>
+): readonly TermOperator[] {
+  // If the token is invalid we want to use the possible expected types as our filter type
+  const validTypes = filterToken.invalid?.expectedType ?? [filterToken.filter];
+
+  // Determine any interchangable filter types for our valid types
+  const interchangeableTypes = validTypes.map(
+    type => interchangeableFilterOperators[type] ?? []
+  );
+
+  // Combine all types
+  const allValidTypes = [...new Set([...validTypes, ...interchangeableTypes.flat()])];
+
+  // Find all valid operations
+  const validOps = new Set<TermOperator>(
+    allValidTypes.map(type => filterTypeConfig[type].validOps).flat()
+  );
+
+  return [...validOps];
+}
+
+export function isWithinToken(node: {location: LocationRange}, position: number) {
```

Comment:
Actually can we make this part of the searchSyntax/utils? I needed something like this in the highlighting component that's part of that module.